lib/cisco_node_utils/vni.rb

```diff
@@ -232,16 +244,21 @@ def mapped_vlan
     end
 
     def mapped_vlan=(vlan)
-      if vlan == default_vlan
+      if vlan == default_mapped_vlan
         state = 'no'
         vlan = mapped_vlan
       else
         state = ''
       end
-      config_set('vni', 'mapped_vlan', vlan: vlan, state: state, vni: @vni_id)
+      # Nothing to be done if vlan is nil
```

Comment:
This should raise an error if vlan is nil. I would just remove the vlan.nil? check altogether.